### DISCUSSION TITLE
legacy sigalg disallowed or unsupported

### DISCUSSION THREAD

**Discussion by eugenestepaniuk** [sequelize/sequelize#15899] [Help & Questions]
I have an api on nodejs and when I try to connect to the database I get an error: _"(node:14320) [DEP0123] DeprecationWarning: Setting the TLS ServerName to an IP address is not permitted by RFC 6066. This will be ignored in a future version.
(Use `node --trace-deprecation ...` to show where the warning was created)
Failed to connect to ip:port - F8510000:error:0A00014D:SSL routines:tls_process_key_exchange:legacy sigalg disallowed or unsupported:c:\ws\deps\openssl\openssl\ssl\statem\statem_clnt.c:2263:"_
My connection:
```
export default new Sequelize({
  dialect: "mssql",
  database: environment.database.database,
  host: environment.database.host,
  port: environment.database.port,
  username: environment.database.username,
  password: environment.database.password,
  logging: !environment.production ? console.log : false,
  dialectOptions: {
    options: {
      // encrypt: false,
      enableArithAbort: true,
      cryptoCredentialsDetails: {
        minVersion: "TLSv1",
      },
    },
  },
  define: {
    freezeTableName: true,
    timestamps: false,
  },
});
```
The connection was initially without encrypt in dialectOptions, I found information on the Internet if you make encrypt: false then the problem may disappear, if I uncomment this line the problem really disappears
The question is, how safe and generally normal is it to do so? Could you suggest other solutions to the error?

upd. this error started to occur after updating nodejs from version 16 to 19

**Comment by WikiRik** [comment]
We're passing the dialectOptions as-is to the connector so I think they might be able to help you out there; https://github.com/tediousjs/tedious

**Comment by eugenestepaniuk** [comment]
ok, thanks, I'll contact them
